bundles/org.eclipse.jface/src/org/eclipse/jface/resource/FontRegistry.java

```diff
@@ -207,7 +208,7 @@ void addAllocatedFontsToStale(Font defaultFont) {
 	 */
 	protected Runnable displayRunnable = this::clearCaches;
 
-	private boolean displayDisposeHooked;
+	private final Set<Display> displayDisposeHooked = new HashSet<>();
```

Comment:
may i ask if this can be used concurrently. Especially if one uses two Display-Threads on windows will they share this instance so that it be thread safe?

Reply:
I made this thread safe in 11c2271.
If I am not mistaken, it may not be necessary, because even though threads may see different states of the data structure, only the data that a thread itself has added is relevant. I.e., the `contains` check only needs to see the display that may have been added by the exact same thread before. But that would be hard to understand and the thread-safe solution does also not introduce performance drawbacks.
An alternative might be to use a thread-local boolean indicating whether the display for the current thread was hooked, but I would also find that harder to understand (but that might be up to personal preference).

Reply:
@jukzi has your concern been addressed? May I merge this PR?